### ISSUE TITLE
Persisting current user id and using in a route?

### ISSUE THREAD

**Issue by alex-braun** [ga-wdi-boston/capstone-project#275]
For my project, I have a view that displays all recipes on my page.  Each one of these recipes has an edit and delete button.  Currently, I am getting the user_id of each recipe to see who created it, and I ALREADY have functionality on the backend that prevents anyone except for the user who created it from performing those CRUD actions (delete, put).  

My problem is that these buttons still display for ALL of the recipes despite some of them not being functional for the recipes that are not created by the current user.  

My backend simply does Recipe.all for index, and then requires currentUser for all other actions.

Is it too much to ask for to have:

```
{{#if <function in route> }}
```

 in my template 

AND then possibly reference the user model or variable from 'auth' (which holds user.id) in my route with a simple if user.id !== user_id? 

Lauren told me about Ember.inject.service and I know this involves local storage, but I cannot wrap my head around this.

Long story short, helping me persist a variable globally would be really swell.

**Comment by laurenfazah** [comment]
Yes, very doable.

Credentials are stored in local storage, yes? Which service in your ember app has that access to that credentials object? You could inject that service into a route to access its credentials property.

Think through that and give it a try. Obviously also ask any clarifying questions you have.

One place this is happening that you could reference is `my-application`, where a check for `isAuthenticated` is happening.

**Comment by gaand** [comment]
You could also add a Boolean to the recipe as I suggested. See the example serializer in rails API template.

**Comment by alex-braun** [comment]
OK Antony, I will try that if I can't get this to work.
In my index route I have added the following

```
import Ember from 'ember';

export default Ember.Route.extend({
  auth: Ember.inject.service(),
  userId: Ember.computed.alias('auth.credentials.id'),

```

still trying to figure out where to make the comparison between userId and recipes.user_id.

**Comment by gaand** [comment]
My suggestion will be a lot less code and much easier to debug.  Just saying.

**Comment by alex-braun** [comment]
I went with Antony's suggestion.
In my rails recipe serializer, an instance method `editable` evaluates to `true` if the current scope is equal to the user who created each recipe object. 

```
class RecipeSerializer < ActiveModel::Serializer
  attributes :id, :name, :category, :ingredients, :directions, :imageurl, :user_id, :editable
  has_one :user

  def editable
    scope == object.user
  end
end
```

In my Ember recipe model:

```
export default DS.Model.extend({
  name: DS.attr('string'),
  category: DS.attr('string'),
  directions: DS.attr('string'),
  ingredients: DS.attr('string'),
  imageurl: DS.attr('string'),
  user_id: DS.attr('number'),
  user: DS.attr(),
  editable: DS.attr('boolean'),
```

finally, in the template:

```
{{#if recipe.editable}}
  (buttons)
{{/if}}
```

Works beautifully.
